### PR TITLE
ASoC: Intel: sof-sdw: add TGL_HDMI quirk for all TigerLake devices.

### PR DESCRIPTION
**Pull request by plbossart**

### CODE REVIEW

**Review by RanderWang — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

File: sound/soc/intel/boards/sof_sdw.c [hunk 1]
```diff
@@ -130,18 +120,21 @@ static const struct dmi_system_id sof_sdw_quirk_table[] = {
...
-			DMI_MATCH(DMI_PRODUCT_NAME, "Ice Lake Client"),
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc"),
+			DMI_EXACT_MATCH(DMI_PRODUCT_SKU, "0A3E")
 		},
-		.driver_data = (void *)SOF_SDW_PCH_DMIC,
+		.driver_data = (void *)(SOF_RT711_JD_SRC_JD2 |
+					SOF_RT715_DAI_ID_FIX),
 	},
-	{
+		{
```
Comment (lyakh) [inline sound/soc/intel/boards/sof_sdw.c]: this seems to be an accidental change here, looks like a copy-paste

Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c]: bad git merge, it was removed and re-added. fixed now.

**Review by kv2019i — DISMISSED** [review]

**Review by bardliao — DISMISSED** [review]

**Review by lyakh — DISMISSED** [review]

**Review by RanderWang — DISMISSED** [review]

**Comment by plbossart** [comment]
@kv2019i there's actually a problem here. If we add 4 HDMI devices in topologies, and the topologies are updated before this commit is added to the distro kernel then the topology will fail to load. In other words, if someone updated sof-bin to v1.7 or 1.8, they would need an updated kernel.

**Comment by kv2019i** [comment]
> @kv2019i there's actually a problem here. If we add 4 HDMI devices in topologies, and the topologies are updated before this commit is added to the distro kernel then the topology will fail to load. In other words, if someone updated sof-bin to v1.7 or 1.8, they would need an updated kernel.

@plbossart Yeah. We had the brief thread about bringing some UCMv2 like conditionally to topology descriptions last year, and this at the root of problem. 

Do you think this is acceptable for the sdw machines? For generic hda mach driver, I'm not comfortable with this as so many systems are possibly affected.

**Comment by plbossart** [comment]
@ranj063 can you enlighten us on how the match between topology and machine driver is done? 

Is there a way we could tag a dailink as 'optional' so that if somehow the topology load failed on that dailink we ignore it.

**Comment by plbossart** [comment]
@kv2019i we have a topology with a dedicated 'tgl' prefix so there's no issue to add this 4th display only for TGL, and I added the quirks as fixes, so they will be added to the stable kernels likely before the sof-bin for SoundWire is updated and pushed to devices.

**Review by lyakh — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
Ack looks good.

**Pull request merged**